src/areaCalculator/AreaCalculator.java

```diff
@@ -1,11 +1,43 @@
 package areaCalculator;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class AreaCalculator {
 
-    public double calculateAreaOfRectangle(Rectangle rectangle) {
-        double area = rectangle.getLength() * rectangle.getWidth();
-        System.out.println("Area of rectangle (" + rectangle.getLength() + " x "
-                + rectangle.getWidth() + ") = " + area);
-        return area;
+    private List<Shape> shapeList;
+
+    public AreaCalculator() {
+        shapeList = new ArrayList<>();
+    }
+
+    void addShapeToList(Shape shape) {
+        shapeList.add(shape);
+    }
+
+    public void calculateArea() {
+        for (Shape shape : shapeList) {
+            if (shape instanceof Rectangle) {
+                calculateRectangleArea(shape);
+            }
+            if (shape instanceof Circle) {
+                calculateCircleArea(shape);
+            }
+        }
+    }
+
+    private void calculateRectangleArea(Shape shape) {
```

Comment:
Currently  you allow any shape to be passed to this method, although you know that it can be only Rectangle. 
Do the type casting in the place you pass the argument  to the method  and instead  accept Rectangle here.